vnext/Microsoft.ReactNative/ViewProps.idl

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "DocString.h"
+import "IJSValueReader.idl";
+
+namespace Microsoft.ReactNative {
+  [webhosthidden]
+  [experimental]
+  DOC_STRING("Interface to implement custom view component properties.")
+  interface IComponentProps
+  {
+    // TODO add an exported method to get the matching hash for a propname?
+    // TODO or remove the hash from the interface?
+    DOC_STRING("This method will be called for each property that comes from JavaScript.  It is up to an individual implementation to store the values of properties for access within its UpdateProps.  Properties that are part of ViewProps can be accessed from the @ViewProps object, and so do not need to be stored.")
```

Comment:
>DOC_STRING [](http://example.com/codeflow?start=4&length=10)

nit: it would be nice to avoid long lines of text like this. Please slit up the comment into multiple lines. See IRectContext.idl as an example how to do it.